test/functional/apps/discover/_large_string.js

```diff
@@ -0,0 +1,61 @@
+import expect from 'expect.js';
+
+export default function ({ getService, getPageObjects }) {
+  const esArchiver = getService('esArchiver');
+  const log = getService('log');
+  const retry = getService('retry');
+  const PageObjects = getPageObjects(['common', 'home', 'settings', 'discover']);
+
+  describe('test large strings', function () {
+    before(async function () {
+      await esArchiver.load('empty_kibana');
+      await esArchiver.loadIfNeeded('hamlet');
+      await PageObjects.settings.createIndexPattern('testlargestring', null);
+    });
+
+    it('verify the large string book present', async function () {
+      const ExpectedDoc =
+           'mybook:Project Gutenberg EBook of Hamlet, by William Shakespeare'
+         + ' This eBook is for the use of anyone anywhere in the United States'
+         + ' and most other parts of the world at no cost and with almost no restrictions whatsoever.'
+         + ' You may copy it, give it away or re-use it under the terms of the'
+         + ' Project Gutenberg License included with this eBook or online at www.gutenberg.org.'
+         + ' If you are not located in the United States,'
+         + ' you’ll have to check the laws of the country where you are'
+         + ' located before using this ebook.'
+         + ' Title: Hamlet Author: William Shakespeare Release Date: November 1998 [EBook #1524]'
+         + ' Last Updated: December 30, 2017 Language: English Character set encoding:';
+
+      let rowData;
+      await retry.try(async function tryingForTime() {
+        await PageObjects.common.navigateToApp('discover');
+        rowData = await PageObjects.discover.getDocTableIndex(1);
+        log.debug('rowData.length=' + rowData.length);
+        expect(rowData.substring(0, 200)).to.be(ExpectedDoc.substring(0, 200));
```

Comment:
This seems really brittle to me as we discussed.  It seems like it would break if we were to run this on another browser.  I think we should be asserting the absence of a loading spinner rather than the presence of a specific (and hardcoded) string.

Reply:
@rasroh there is a problem here in that the `navigateToApp('discover')` should not be inside the try loop.  The way it currently is it will re-navigate to Discover every time through the loop which could cause it to never load in time.

@bmcconaghy this try loop is the way all of the discover tests have always worked (in almost 3 years since we first automated UI tests on Discover).  The loading indicator only tells when the request to Elasticsearch has been completed, not when the rendering in Kibana is completed.  There might be something better we could do in all the discover tests but nobody has figured it out yet.